src/devices/cddasongloader.cpp

```diff
@@ -152,6 +156,8 @@ void CddaSongLoader::LoadSongsFromCdda() {
           songs[i++].set_length_nanosec(duration);
         }
       }
+      g_list_free(entries);
```

Comment:
I don't think we're supposed to free the entries list. gst_toc_get_entries documentation shows "transfer none".

Reply:
Ah, I see. I'll revert that. Not entirely familiar with gstreamer and find their docs somewhat confusing at times.

Reply:
the gst_toc_unref seems to be misplaced also .. it should, of course, go into the if(toc) {} body. I'll fix that as well.